### ISSUE TITLE
RAC ListBox preview bug on safari when using a combination of position relative and absolute

### ISSUE THREAD

**Issue by fredericrous** [adobe/react-spectrum#6354]
### Provide a general summary of the issue here

In ListBox component, from RAC, we are locked with a DragPreview that renders inside ListBox.
When ListBox is inside a complex layout using a combination of position relative and absolute, the drag preview, on safari, gets shown dark

### 🤔 Expected Behavior?

The provided DragPreview is rendered in all browsers

### 😯 Current Behavior

The provided DragPreview is rendered buggy in Safari, I see it at a different position than what I expect and the div is dark

### 💁 Possible Solution

- allow us to replace the full preview component, including DragPreview

or

- implement createPortal to render DragPreview outside of the dom

or

- allow us to specify a portal string where the portal would be created

### 🖥️ Steps to Reproduce

- I'm using reactflow
- I render a listbox inside nodes
- I then drag an item from one listbox to another
- drag preview is off on Chrome for IOS and on Safari for Desktop

### Version

1.2.0

### What browsers are you seeing the problem on?

Chrome, Safari

### If other, please specify.

Safari 17.3.1, Chrome for IOS v123.0.6312.52

### What operating system are you using?

macos 12.5, ios 17.4

### 🕷 Tracking Issue

similar issue: https://github.com/adobe/react-spectrum/issues/4797

**Comment by fredericrous** [comment]
@BenBeattieHood I see in the code a property UNSTABLE_portalContainer for modal, tooltip and popover. It looks like drag preview were forgotten

**Comment by snowystinger** [comment]
I don't believe we use `createPortal` for the DragPreview. Instead, a component is rendered offscreen, once it's rendered, the image of that rendered component will be set to the dataTransfer's dragImage.

If I recall, we don't portal it because if we did, then the rendered component wouldn't have any inherited styles.

I don't quite follow the issue, could you provide a codesandbox or stackblitz demonstrating the issue? I'm unclear if this is a duplicate of #4797 linked in the Issue or if there is a different issue.

**Comment by fredericrous** [comment]
Thank you for your message! I believe the same fix might resolve both issues, but the reproduction workflow and the resulting problems appear to be different.

A simpler reproduction scenario for this issue (without involving React Flow) could be to render the list inside a DOM structure like: div.relative > div.absolute > div.relative, with several sibling nodes.

Unfortunately, I won’t be able to test this on a newer version of RAC and Safari anytime soon, as the project where I was using RAC is currently on hold due to other life priorities
